http/oidc/src/main/java/org/wildfly/security/http/oidc/ServerRequest.java

```diff
@@ -274,4 +279,40 @@ public static AccessAndIDTokenResponse getBearerToken(OidcClientConfiguration oi
         }
         return tokenResponse;
     }
+
+    public static String getRequestUri(String request, OidcClientConfiguration deployment) throws OidcException {
+        HttpPost parRequest = new HttpPost(deployment.getPushedAuthorizationRequestEndpoint());
+        List<NameValuePair> formParams = new ArrayList<NameValuePair>();
+        formParams.add(new BasicNameValuePair(REQUEST, request));
+        ClientCredentialsProviderUtils.setClientCredentials(deployment, parRequest, formParams);
+        parRequest.addHeader("Content-type", "application/x-www-form-urlencoded");
+
+        UrlEncodedFormEntity form = new UrlEncodedFormEntity(formParams, StandardCharsets.UTF_8);
+        parRequest.setEntity(form);
+
+        HttpResponse response;
+        try {
+            response = deployment.getClient().execute(parRequest);
+        } catch (Exception e) {
+            throw log.failedToSendPushedAuthorizationRequest(e);
+        }
+        if (response.getStatusLine().getStatusCode() != HttpStatus.SC_CREATED) {
+            EntityUtils.consumeQuietly(response.getEntity());
+            throw log.unexpectedResponseCodeFromOidcProvider(response.getStatusLine().getStatusCode());
+        }
+        try {
+            InputStream inputStream = response.getEntity().getContent();
```

Comment:
Can this method make use of any existing utilities like readString?

Reply:
Done